src/compiler/error-detector.js

```diff
@@ -8,6 +8,8 @@ const prohibitedKeywordRE = new RegExp('\\b' + (
   'super,throw,while,yield,delete,export,import,return,switch,default,' +
   'extends,finally,continue,debugger,function,arguments'
 ).split(',').join('\\b|\\b') + '\\b')
+const deleteExpressionRE = /\bdelete\s*\([^\)]*\)/
```

Comment:
There might be some other unary operators we need to check in theory, like `typeof`, `void` ?